### PR TITLE
End round when all users leave VC [#76]

### PR DESCRIPTION
**Pull request by taahamahdi**
Can't send the song message because there is no way for us to get the text channel

### CODE REVIEW

File: kmq.js [hunk 1]
```diff
@@ -36,6 +36,25 @@ client.on("message", (message) => {
     }
 });
 
+client.on("voiceStateUpdate", (oldMember, newMember) => {
```
Comment (Brainicism) [inline kmq.js]: The parameters to the callback are `oldVoiceState` and `newVoiceState`, not specific to member

File: kmq.js [hunk 2]
```diff
@@ -36,6 +36,25 @@ client.on("message", (message) => {
...
+    let oldUserChannel = oldMember.voiceChannel;
+    let newUserChannel = newMember.voiceChannel;
+    if (newUserChannel === undefined) {
+        // User left voice channel, check if bot is only one left
+        if (oldUserChannel.members.size === 1) {
+            let guildID = oldUserChannel.guild.id;
+            let voiceConnection = client.voiceConnections.get(guildID);
+            if (voiceConnection) {
+                sendSongMessage(message, gameSession, true);
```
Comment (Brainicism) [inline kmq.js]: Where is `message` coming from

File: kmq.js [hunk 3]
```diff
@@ -36,6 +36,25 @@ client.on("message", (message) => {
     }
 });
 
+client.on("voiceStateUpdate", (oldMember, newMember) => {
+    let oldUserChannel = oldMember.voiceChannel;
+    let newUserChannel = newMember.voiceChannel;
```
Comment (Brainicism) [inline kmq.js]: Documentation says that this should be `.channel`, are you sure this works?

File: kmq.js [hunk 4]
```diff
@@ -36,6 +36,25 @@ client.on("message", (message) => {
...
+    if (newUserChannel === undefined) {
```
Comment (Brainicism) [inline kmq.js]: Why is the voicestate undefined? I'm struggling to understand how this works, especially given that the bot is still in the voice channel.

File: kmq.js [hunk 5]
```diff
@@ -36,6 +36,25 @@ client.on("message", (message) => {
...
+        // User left voice channel, check if bot is only one left
+        if (oldUserChannel.members.size === 1) {
+            let guildID = oldUserChannel.guild.id;
+            let voiceConnection = client.voiceConnections.get(guildID);
```
Comment (Brainicism) [inline kmq.js]: Voicestate should contain the corresponding connection already

File: kmq.js (hunk 3 above)
Comment (taahamahdi) [reply kmq.js]: Yup! I tried with `.channel` and it doesn't work.

**Pull request merged**